cxx/gendb.hh

```diff
@@ -0,0 +1,65 @@
+// Copyright 2024
+// See LICENSE.txt
+
+#pragma once
+#include <map>
+#include <random>
+#include <string>
+
+#include "distributions/crp.hh"
+#include "hirm.hh"
+#include "observations.hh"
+#include "pclean/schema.hh"
+#include "pclean/schema_helper.hh"
+
+class GenDB {
+ public:
+  const PCleanSchema& schema;
+
+  // TODO(emilyaf): Merge PCleanSchemaHelper and GenDB.
+  PCleanSchemaHelper schema_helper;
+
+  // This data structure contains entity sets and linkages. Semantics are
+  // map<class_name, map<primary_key, map<reference_field_name, ref_val>>>,
+  // where primary_key and ref_val are (integer) entity IDs.
+  std::map<std::string, std::map<int, std::map<std::string, int>>>
+      reference_values;
+
+  HIRM* hirm;
```

Comment:
Probably worth commenting here that this class owns hirm.

Reply:
Done.